src/core/config/config.rs

```diff
@@ -523,6 +523,55 @@ impl Config {
         types
     }
 
+    pub fn interfaces_types_map(&self) -> HashMap<String, HashSet<String>> {
+        let mut interfaces_types: HashMap<String, HashSet<String>> = HashMap::new();
+
+        for (type_name, type_definition) in self.types.iter() {
+            for implement_name in type_definition.implements.clone() {
+                match interfaces_types.get_mut(&implement_name) {
+                    Some(types_set) => {
+                        types_set.insert(type_name.clone());
+                    }
+                    None => {
+                        let mut types_set = HashSet::new();
+                        types_set.insert(type_name.clone());
+                        interfaces_types.insert(implement_name, types_set);
+                    }
+                }
+            }
+        }
+
+        fn recursive_interface_type_merging(
+            types_set: &HashSet<String>,
+            interfaces_types: &HashMap<String, HashSet<String>>,
+        ) -> HashSet<String> {
+            let mut types_set_local = HashSet::new();
+
+            for type_name in types_set.iter() {
+                match interfaces_types.get(type_name) {
+                    Some(types_set_inner) => {
+                        let types_set_inner =
+                            recursive_interface_type_merging(types_set_inner, interfaces_types);
+                        types_set_local = types_set_local.merge_right(types_set_inner);
+                    }
+                    None => {
+                        types_set_local.insert(type_name.to_string());
+                    }
+                }
+            }
+
+            types_set_local
+        }
```

Comment:
Good catch, infinite recursion have not been tested

Reply:
```gql
type Query {
  data: A
}

type A implements T1 & T3 & T4 {
  id: ID!
  foo: String
}

type B implements T2 {
  id: ID!
  bar: String
}

interface T1 implements T4 & T3 {
  foo: String
}

interface T2 {
  bar: String
}

interface T3 implements T1 & T4 {
  foo: String
}

interface T4 implements T3 & T1 {
  foo: String
}
```

If you use this schema you will get an error indicating that Interfaces cannot reference each other, so there is no fear of infinity toops

Reply:
@meskill its hard to follow the same approach as `src/core/config/transformer/nested_unions.rs` because I do not store something in the `Config` object to implement a transformer on `Config`. But I can place it in a separate file. Will this work?

Reply:
could you please check that we have a test for this? And please verify the recursion case against json/yaml config also, similar to `tests/execution/test-interface-from-json.md`

about the code itself - I see the issue with the transformer, let's stick with your current approach, it's fine